src/System.Runtime.Extensions/tests/System/EnvironmentTests.cs

```diff
@@ -267,57 +267,65 @@ public void GetFolderPath_OSX_NonEmptyFolderPaths(Environment.SpecialFolder fold
         // [InlineData(Environment.SpecialFolder.CommonOemLinks)]
-        [InlineData(Environment.SpecialFolder.CommonPictures)]
-        [InlineData(Environment.SpecialFolder.CommonStartMenu)]
-        [InlineData(Environment.SpecialFolder.CommonPrograms)]
-        [InlineData(Environment.SpecialFolder.CommonStartup)]
-        [InlineData(Environment.SpecialFolder.CommonDesktopDirectory)]
-        [InlineData(Environment.SpecialFolder.CommonTemplates)]
-        [InlineData(Environment.SpecialFolder.CommonVideos)]
-        [InlineData(Environment.SpecialFolder.Fonts)]
-        [InlineData(Environment.SpecialFolder.NetworkShortcuts)]
+        [InlineData(Environment.SpecialFolder.CommonPictures, false)]
+        [InlineData(Environment.SpecialFolder.CommonStartMenu, false)]
+        [InlineData(Environment.SpecialFolder.CommonPrograms, false)]
+        [InlineData(Environment.SpecialFolder.CommonStartup, false)]
+        [InlineData(Environment.SpecialFolder.CommonDesktopDirectory, false)]
+        [InlineData(Environment.SpecialFolder.CommonTemplates, false)]
+        [InlineData(Environment.SpecialFolder.CommonVideos, false)]
+        [InlineData(Environment.SpecialFolder.Fonts, true)]
+        [InlineData(Environment.SpecialFolder.NetworkShortcuts, false)]
         // [InlineData(Environment.SpecialFolder.PrinterShortcuts)]
-        [InlineData(Environment.SpecialFolder.UserProfile)]
-        [InlineData(Environment.SpecialFolder.CommonProgramFilesX86)]
-        [InlineData(Environment.SpecialFolder.ProgramFilesX86)]
-        [InlineData(Environment.SpecialFolder.Resources)]
+        [InlineData(Environment.SpecialFolder.UserProfile, false)]
+        [InlineData(Environment.SpecialFolder.CommonProgramFilesX86, true)]
+        [InlineData(Environment.SpecialFolder.ProgramFilesX86, true)]
+        [InlineData(Environment.SpecialFolder.Resources, true)]
         // [InlineData(Environment.SpecialFolder.LocalizedResources)]
-        [InlineData(Environment.SpecialFolder.SystemX86)]
-        [InlineData(Environment.SpecialFolder.Windows)]
+        [InlineData(Environment.SpecialFolder.SystemX86, true)]
+        [InlineData(Environment.SpecialFolder.Windows, true)]
         [PlatformSpecific(TestPlatforms.Windows)]  // Tests OS-specific environment
-        public unsafe void GetFolderPath_Windows(Environment.SpecialFolder folder)
+        public unsafe void GetFolderPath_Windows(Environment.SpecialFolder folder, bool existsInAppContainer = true)
         {
             string knownFolder = Environment.GetFolderPath(folder);
-            Assert.NotEmpty(knownFolder);
+            // If you are not executing inside an AppContainer or the folder is allowed in it the path must not be empty.
+            if (!PlatformDetection.IsWinRT || existsInAppContainer)
+            {
+                Assert.NotEmpty(knownFolder);
+            }
+            else
+            {
+                Assert.Empty(knownFolder);
```

Comment:
`// if you hit this assert it means that SHGetFolderPath inside a UWP has been fixed. Check the value is reasonable and update the test data to expect a path from now on. `